python/metric_pipeline_utils/coord_util.py

```diff
@@ -0,0 +1,120 @@
+    The intent is to use this for a set of measurements of the same source
+    but that's neither enforced nor required.
+    Parameters
+    ----------
+    cat : collection
+         Object with .get method for 'coord_ra', 'coord_dec' that returns radians.
+    Returns
+    -------
+    ra_mean : `float`
+        Mean RA in radians.
+    """
+    meanRa, meanDec = averageRaDecFromCat(cat)
+    return meanRa
+
+
+def averageDecFromCat(cat):
+    """Compute the average declination from a catalog of measurements.
+    This function is used as an aggregate function to extract just declination
+    from a lsst.afw.table.MultiMatch final match catalog.
+    The actual computation involves both RA and Dec.
+    The intent is to use this for a set of measurements of the same source
+    but that's neither enforced nor required.
+    Parameters
+    ----------
+    cat : collection
+         Object with .get method for 'coord_ra', 'coord_dec' that returns radians.
+    Returns
+    -------
+    dec_mean : `float`
+        Mean Dec in radians.
+    """
+    meanRa, meanDec = averageRaDecFromCat(cat)
+    return meanDec
+
+
+def averageRaDecFromCat(cat):
+    """Calculate the average right ascension and declination from a catalog.
+    Convenience wrapper around averageRaDec
+    Parameters
+    ----------
+    cat : collection
+         Object with .get method for 'coord_ra', 'coord_dec' that returns radians.
+    Returns
+    -------
+    ra_mean : `float`
+        Mean RA in radians.
+    dec_mean : `float`
+        Mean Dec in radians.
+    """
+    return averageRaDec(cat.get('coord_ra'), cat.get('coord_dec'))
+
+
+def averageRaDec(ra, dec):
+    """Calculate average RA, Dec from input lists using spherical geometry.
+    Parameters
+    ----------
+    ra : `list` [`float`]
+        RA in [radians]
+    dec : `list` [`float`]
+        Dec in [radians]
+    Returns
+    -------
+    float, float
+       meanRa, meanDec -- Tuple of average RA, Dec [radians]
+    """
+    assert(len(ra) == len(dec))
+
+    angleRa = [geom.Angle(r, geom.radians) for r in ra]
+    angleDec = [geom.Angle(d, geom.radians) for d in dec]
+    coords = [geom.SpherePoint(ar, ad, geom.radians) for (ar, ad) in zip(angleRa, angleDec)]
+
+    meanRa, meanDec = geom.averageSpherePoint(coords)
+
+    return meanRa.asRadians(), meanDec.asRadians()
+
+
+def sphDist(ra_mean, dec_mean, ra, dec):
+    """Calculate distance on the surface of a unit sphere.
+    Parameters
+    ----------
+    ra_mean : `float`
+        Mean RA in radians.
+    dec_mean : `float`
+        Mean Dec in radians.
+    ra : `numpy.array` [`float`]
+        Array of RA in radians.
+    dec : `numpy.array` [`float`]
+        Array of Dec in radians.
+    Notes
+    -----
+    Uses the Haversine formula to preserve accuracy at small angles.
+    Law of cosines approach doesn't work well for the typically very small
+    differences that we're looking at here.
+    """
+    # Haversine
+    dra = ra - ra_mean
```

Comment:
I believe the `geom` code in the stack uses the Haversine formula.  It's a little unfortunate that we reimplement it here.  We don't need to change it here, but I'd like an issue in the non-PoC epic to track it.